controller/oidc_auth/negotiate.go

```diff
@@ -49,17 +88,71 @@ func negotiateBodyContentType(r *http.Request) (string, *errorz.ApiError) {
 	}
 }
 
-// parseAcceptHeader parses HTTP accept headers and returns an array of supported
-// content types sorted by quality factor (0=most desired response type). The return
-// strings are the content type only (e.g. "application/json")
-func parseAcceptHeader(acceptHeader string) []string {
-	parts := strings.Split(acceptHeader, ",")
-	contentTypes := make([]string, len(parts))
+// AcceptEntry represents a parsed Accept header entry
+type AcceptEntry struct {
+	FullHeader string  // Original full header
+	MediaType  string  // Parsed media type
+	QValue     float64 // Quality factor (default 1.0)
+}
+
+// Match checks if a given media type matches this AcceptEntry, supporting wildcards.
+func (a AcceptEntry) Match(mediaType string) bool {
+	// Split main and subtype (e.g., "image/png" -> "image", "png")
+	expectedParts := strings.SplitN(a.MediaType, "/", 2)
+	incomingParts := strings.SplitN(mediaType, "/", 2)
+
+	if len(expectedParts) != 2 || len(incomingParts) != 2 {
+		return false // Invalid format
+	}
+
+	if a.MediaType == "*/*" {
+		return true // Matches anything
+	}
+
+	// main type and sup type match exactly or *
+	return (expectedParts[0] == incomingParts[0] || expectedParts[0] == "*") &&
+		(expectedParts[1] == incomingParts[1] || expectedParts[1] == "*")
+}
+
+// parseAcceptHeader parses an Accept header string and returns a sorted slice of AcceptEntry based on q value (high to low)
+func parseAcceptHeader(header string) []AcceptEntry {
+	if header == "" {
+		return nil
+	}
+
+	entries := strings.Split(header, ",")
+	var acceptHeaders []AcceptEntry
+
+	for _, entry := range entries {
+		entry = strings.TrimSpace(entry)
+		parts := strings.Split(entry, ";")
+
+		mediaType := strings.TrimSpace(parts[0])
+		qValue := 1.0 // Default q-value
 
-	for i, part := range parts {
-		typeAndFactor := strings.Split(strings.TrimSpace(part), ";")
-		contentTypes[i] = typeAndFactor[0]
+		// Check if there's a q= parameter
+		if len(parts) > 1 {
```

Comment:
does this assume that if there's only a single `q param` that the value will always be 1 even if only "q=10" is supplied? why not just range over the parts looking for q=?

Reply:
Per the spec, there is one `q` per content type, and if it isn't specified,d it is defaulted to 1.